sound/soc/soc-pcm.c

```diff
@@ -2803,9 +2827,18 @@ static int soc_get_playback_capture(struct snd_soc_pcm_runtime *rtd,
 				cpu_dai = asoc_rtd_to_cpu(rtd, 0);
 			} else if (rtd->dai_link->num_cpus == rtd->dai_link->num_codecs) {
 				cpu_dai = asoc_rtd_to_cpu(rtd, i);
+			} else if (rtd->dai_link->codec_ch_maps) {
+				int cpu_id = rtd->dai_link->codec_ch_maps[i].connected_cpu_id;
+
+				cpu_dai = asoc_rtd_to_cpu(rtd, cpu_id);
+			} else if (rtd->dai_link->num_cpus > rtd->dai_link->num_codecs) {
+				dev_err(rtd->card->dev,
+					"codecs %d must be equal or larger than cpus %d\n",
+					rtd->dai_link->num_codecs, rtd->dai_link->num_cpus);
+				return -EINVAL;
 			} else {
 				dev_err(rtd->card->dev,
-					"N cpus to M codecs link is not supported yet\n");
+					"N cpus to M codecs link is not supported: no mapping provided\n");
 				return -EINVAL;
 			}
```

Comment:
use __func__ and add "or codec number < cpu number"

Reply:
done